File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        map = {}

        for i in nums:
            if i in map:
                map[i] = map[i]+1
            else:
                map[i] = 1

        for key in map:
            if map[key] > 1:
                return key           

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer,Integer> mpp=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int freq=mpp.getOrDefault(nums[i],0);
            mpp.put(nums[i],freq+1);
        }
        int element=-1;
        for(Map.Entry<Integer,Integer> entry:mpp.entrySet()){
            if(entry.getValue()>1){
                element=entry.getKey();
            }
        }
        return element;
    }
}